plugins/modules/cloud/misc/terraform.py

```diff
@@ -171,9 +182,23 @@ def preflight_validation(bin_path, project_path, variables_args=None, plan_file=
 
     rc, out, err = module.run_command([bin_path, 'validate'] + variables_args, cwd=project_path, use_unsafe_shell=True)
     if rc != 0:
+        with open(dir_name, 'a+', newline='') as file_op:
```

Comment:
This should only happen if `dir_name` is not `None`. Also, better to use `log_error_path` here and remove `dir_name` from this function.